### DISCUSSION TITLE
Winforma for .NET core

### DISCUSSION THREAD

**Discussion by vvijayalakshmi21** [dotnet/winforms#4628] [Q&A] [answered]
Dear all, 
A newbie to .net core migration here. 
We've a winforms application that we're looking to migrate to .net core
 
Does a windows forms application targeting .net core run on a Linux or Mac platform?

If no, is there any alternative?

**Comment by RussKie** [comment, marked as answer]
Howdy,

> Does a windows forms application targeting .net core run on a Linux or Mac platform?

Windows Forms apps generally run only on the Windows platform (hence the name 😉). 

That's said, Windows Forms .NET Framework app can be run under [Mono WinForms](https://www.mono-project.com/docs/gui/winforms/) on Linux and MacOS. I've heard reports Windows Forms .NET apps can run under [WineHQ](https://www.winehq.org/) as well.
Please note: 1) YMMV, and 2) we don't provide support for these scenarios.



> If no, is there any alternative?

You have a plenty of options to chose from: [.NET MAUI](https://github.com/dotnet/maui), [Blazor](https://dotnet.microsoft.com/apps/aspnet/web-apps/blazor), etc.
